logstash-core/src/main/java/org/logstash/ObjectMappers.java

```diff
@@ -42,6 +43,7 @@
 
     private static final SimpleModule CBOR_DESERIALIZERS =
         new SimpleModule("CborRubyDeserializers")
+            .addDeserializer(RubyString.class, new RubyStringDeserializer())
```

Comment:
Why are there changes to serialization in this PR? Was this to fix test problems?

Reply:
@andrewvc nah, this is kind of an annoying corner case, long story short:

`RubyFixnum` and other Ruby type's `compareTo` method works fine with Java numbers (since those just use `toLong()` and such below the surface), but for `String` and `RubyString` this isn't the case (due to encoding issues) => without adding a `Deserializer` that gives us straight `RubyString` we'd have to add a special code path for comparisons to `String/RubyString` which would've had quite the negative performance impact => `this` :)